src/main/python/deep_qa/training/trainer.py

```diff
@@ -42,6 +42,13 @@ def __init__(self, params: Dict[str, Any]):
             parent_directory = os.path.dirname(self.model_prefix)
             os.makedirs(parent_directory, exist_ok=True)
 
+        # Preferred backend to use for training. If a different backend is detected, we still
+        # train but we also warn the user.
+        self.preferred_backend = params.pop('preferred_backend', None)
+        if self.preferred_backend and self.preferred_backend != K.backend():
+            logger.warning("Preferred backend is %s, but current backend is %s.",
```

Comment:
Is it worth waiting here for user input?  If not I would probably take several lines to display this warning, making it really bold and obvious.

Also, should you update your example experiment with the ASReader to include this parameter?